src/dialog/dialog.service.ts

```diff
@@ -146,17 +148,18 @@ export class DialogService {
 			}
 		});
 
-		// keep track of all dialog subscriptions
-		DialogService.dialogCloseSubscription.add(closeSubscription);
+		// Adds current close subscription to the reference of all close subscriptions for
+		// local dialog service.
+		this.dialogSubscription.add(closeSubscription);
+
+		// keep track of all dialog subscriptions globally.
+		DialogService.dialogCloseSubscription.add(this.dialogSubscription);
```

Comment:
One note, this should be called in the service constructor, since we don't want to accidentally add the subscription again (I don't think it will, but that's an implementation detail of rxjs we shouldn't rely on)